src/main/java/com/github/calebwhiting/runelite/plugins/actionprogress/detect/StringJewellerySpellDetector.java

```diff
@@ -0,0 +1,51 @@
+package com.github.calebwhiting.runelite.plugins.actionprogress.detect;
+
+import com.github.calebwhiting.runelite.data.Magic;
+import com.github.calebwhiting.runelite.plugins.actionprogress.Action;
+import com.github.calebwhiting.runelite.plugins.actionprogress.ActionProgressConfig;
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import net.runelite.api.*;
+import net.runelite.api.events.MenuOptionClicked;
+import net.runelite.client.eventbus.Subscribe;
+
+@Singleton
+public class StringJewellerySpellDetector extends ActionDetector
+{
+	@Inject private ActionProgressConfig config;
+
+	@Inject private Client client;
+
+	@Subscribe
+	public void onMenuOptionClicked(MenuOptionClicked evt)
+	{
+		if (!this.config.magicStringJewellery()) {
+			return;
+		}
+		if (evt.getMenuAction() != MenuAction.CC_OP) {
+			return;
+		}
+		ItemContainer inventory = this.client.getItemContainer(InventoryID.INVENTORY);
+		if (inventory == null) {
+			return;
+		}
+		if (!evt.getMenuTarget().equals("<col=00ff00>String Jewellery</col>")) {
```

Comment:
This seemed like a dirty solution, but I wasn't able to find something nicer for it. Suggestions for improvement (and links to relevant docs) are very welcome